### ISSUE TITLE
Standardise docs

### ISSUE THREAD

**Issue by ben18785** [pints-team/pints#923]
Some of the docs are missing algorithm details (e.g. Remi's method and random walk Metropolis). I'd like these to be there as a) makes it easier for users to understand them and b) makes it easier for us to know what we've done as sometimes these deviate from the methods in the papers and c) makes writing pseudocode much easier in papers (which I have to do anyway, so not wasted time).

Also, there's a mix of math and code types used in the docstrings, so would be good to make consistent.

**Comment by MichaelClerx** [comment]
So...

- [ ] Check that all optimisers & samplers have a reference
- [ ] Ensure same reference style & formatting is used everywhere. Ideally something that renders nicely, which might depend on #849 being implemented first
- [ ] Check all maths is rendered correctly
- [ ] Check we use tex when talking maths, code when talking variable names, implementation details etc.

Anything else?

**Comment by MichaelClerx** [comment]
Prioritising: Should be easily done

**Comment by MichaelClerx** [comment]
See also #849

**Comment by MichaelClerx** [comment]
@ben18785 I think it'd be better to close this and handle each issue we find in the docs as a separate ticket
